src/components/forms/CreatePostForm.js

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from "react";
+import { getCategories } from "../../managers/categoryManager.js";
+import { addPost } from "../../managers/postManager.js";
+
+export const CreatePostForm = (currentUser) => {
+  const [post, setPost] = useState({
+    user_id: currentUser.id,
```

Comment:
the key name on current user that you want to access is confusingly named token, not id.
Update this line to be, user_id: currentUser.token,